### ISSUE TITLE
Loss is always 0 for multiclass problem

### ISSUE THREAD

**Issue by AngledLuffa** [mathiaszinnen/focal_loss_torch#10]
I tried using `FocalLoss(gamma=2.0)` as a drop-in replacement for `nn.CrossEntropyLoss()` in a couple different multiclass problems, and in both cases I got losses of 0.  In one case there are 3 classes, in another there were 20.  Is there a recommended way to use it in that context where I wouldn't expect this to happen?

The context in question is either a sentiment analysis tool or a parser, both part of Stanford's Stanza library:

https://github.com/stanfordnlp/stanza/tree/con_focal

(`con` in this case being short for `constituency`)

If you're interested in digging into this particular problem, I can provide a few datafiles which are public domain

**Comment by msalhab96** [comment]
to use FocalLoss as nn.CrossEntropyLoss() gamma should equal to 0

**Comment by msalhab96** [comment]
If you can give more details to dig deeper that would be better!

**Comment by AngledLuffa** [comment]
I may have not expressed the problem well.  If I run `CrossEntropyLoss` as the loss function, and I print out the first few losses, I get

```
1.1091877222061157
1.0973833799362183
1.0567669868469238
1.1430968046188354
1.0842665433883667
1.1735007762908936
1.1027023792266846
```

So what I wanted was to use the focal loss algorithm instead of cross entropy loss.  I would expect that it is sufficient to do this

```
        loss_function = FocalLoss(gamma=args.loss_focal_gamma)
```

instead of

```
        loss_function = nn.CrossEntropyLoss()
```

However, I then get `nan` for the first few losses:

```
nan
nan
nan
nan
nan
```

This happens regardless of the `gamma` I use.

Like I mentioned, I can easily send a dataset which will demonstrate the error, if that's helpful.

**Comment by msalhab96** [comment]
Thanks for the elaboration, 

are you sure that the predictions in one of the following shapes?
````
(B,)  # for binary classification  
(B, C)  # for multi-class classificatio
(B, M, C)  # for sequene prediction
```

And make sure that all the values are between 0-1

It would be better if you could dump the values you are passing the focal loss in pickle file to try it out

**Comment by msalhab96** [comment]
@AngledLuffa Could you please check this pull request if it throughs an error after passing your input? 

https://github.com/mathiaszinnen/focal_loss_torch/pull/12

**Comment by AngledLuffa** [comment]
Thanks for looking into this.  I suspect that's exactly what is happening
here.  CrossEntropyLoss expects unnormalized input, so I hadn't passed in
softmax output.  I'll have time to check later this afternoon

**Comment by AngledLuffa** [comment]
Yes, that was exactly it.  Thanks for the help.
